python/semantic_kernel/agents/open_ai/assistant_content_generation.py

```diff
@@ -0,0 +1,192 @@
+    message: "ChatMessageContent",
+    allowed_message_roles: list[str] = [AuthorRole.USER, AuthorRole.ASSISTANT],
+) -> "Message":
+    """Class method to add a chat message, callable from class or instance.
+
+    Args:
+        client: The client to use for creating the message.
+        thread_id: The thread id.
+        message: The chat message.
+        allowed_message_roles: The allowed message roles.
+
+    Returns:
+        Message: The message.
+    """
+    if message.role.value not in allowed_message_roles:
+        raise AgentExecutionException(
+            f"Invalid message role `{message.role.value}`. Allowed roles are {allowed_message_roles}."
+        )
+
+    message_contents: list[dict[str, Any]] = get_message_contents(message=message)
+
+    return await client.beta.threads.messages.create(
+        thread_id=thread_id,
+        role=message.role.value,  # type: ignore
+        content=message_contents,  # type: ignore
+    )
+
+
+def get_message_contents(message: "ChatMessageContent") -> list[dict[str, Any]]:
+    """Get the message contents.
+
+    Args:
+        message: The message.
+    """
+    contents: list[dict[str, Any]] = []
+    for content in message.items:
+        if isinstance(content, TextContent):
+            contents.append({"type": "text", "text": content.text})
+        elif isinstance(content, ImageContent) and content.uri:
+            contents.append(content.to_dict())
+        elif isinstance(content, FileReferenceContent):
+            contents.append({
+                "type": "image_file",
+                "image_file": {"file_id": content.file_id},
+            })
+    return contents
+
+
+def generate_message_content(assistant_name: str, message: "Message") -> ChatMessageContent:
+    """Generate message content."""
+    role = AuthorRole(message.role)
+
+    content: ChatMessageContent = ChatMessageContent(role=role, name=assistant_name)  # type: ignore
+
+    for item_content in message.content:
+        if item_content.type == "text":
+            assert isinstance(item_content, TextContentBlock)  # nosec
+            content.items.append(
+                TextContent(
+                    text=item_content.text.value,
+                )
+            )
+            for annotation in item_content.text.annotations:
+                content.items.append(generate_annotation_content(annotation))
+        elif item_content.type == "image_file":
+            assert isinstance(item_content, ImageFileContentBlock)  # nosec
+            content.items.append(
+                FileReferenceContent(
+                    file_id=item_content.image_file.file_id,
+                )
+            )
+    return content
+
+
+def generate_function_call_content(agent_name: str, fccs: list[FunctionCallContent]) -> ChatMessageContent:
+    """Generate function call content."""
+    function_call_content: ChatMessageContent = ChatMessageContent(role=AuthorRole.TOOL, name=agent_name)  # type: ignore
```

Comment:
why not just set items to fccs?
```suggestion
    return ChatMessageContent(role=AuthorRole.TOOL, name=agent_name, items=fccs)
```

Reply:
done - good catch